rest/data/version.go

```diff
@@ -304,6 +308,57 @@ func addFailedAndStartedTests(rows map[string]restModel.BuildList, failedAndStar
 	return nil
 }
 
+func (vc *DBVersionConnector) CreateVersionFromConfig(projectID string, config []byte, user *user.DBUser, message string, active bool) (*version.Version, error) {
+	ref, err := model.FindOneProjectRef(projectID)
+	if err != nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    "error finding project",
+		}
+	}
+	if ref == nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("project %s does not exist", projectID),
+		}
+	}
+	project := &model.Project{}
+	err = model.LoadProjectInto(config, projectID, project)
+	if err != nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("error parsing project config: %s", err.Error()),
+		}
+	}
+	metadata := repotracker.VersionMetadata{
+		IsAdHoc: true,
+		User:    user,
+		Message: message,
+	}
+	newVersion, err := repotracker.CreateVersionFromConfig(ref, project, metadata, false, nil)
+	if err != nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("error creating version: %s", err.Error()),
+		}
+	}
+
+	if active {
+		catcher := grip.NewBasicCatcher()
+		for _, b := range newVersion.BuildIds {
+			catcher.Add(model.SetBuildActivation(b, true, evergreen.DefaultTaskActivator))
```

Comment:
I think SetBuildActivation may be expensive, since it gets dependencies of tasks in each build. Could you use the equivalent of what the UI does for activating an entire version?

Reply:
Both the UI and repotracker call this. We could maybe add a flag for ad hoc/patches that don't check dependencies?

Reply:
I think that's reasonable.

Reply:
added